From 0c7a8aa59e82e0325634591268e66353915daf4b Mon Sep 17 00:00:00 2001
Message-Id: <patch-7.9-0c7a8aa59e8-20230202T091827Z-avarab@gmail.com>
In-Reply-To: <cover-0.9-00000000000-20230202T091827Z-avarab@gmail.com>
References: <230131.86357rrtsg.gmgdl@evledraar.gmail.com>
	<cover-0.9-00000000000-20230202T091827Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Wed, 1 Feb 2023 13:01:28 +0100
Subject: [PATCH 7/9] test-lib.sh: add a lazy GZIP prerequisite
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Move the "gzip --version" lazy prerequisite added in [1] and
copy/pasted to another test in [2] to test-lib.sh. A subsequent commit
will add a third user, let's first stop duplicating it.

1. 96174145fc3 (t5000: simplify gzip prerequisite checks, 2013-12-03)
2. 6c213e863ae (http-backend: respect CONTENT_LENGTH for receive-pack,
   2018-07-27)

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 t/t5000-tar-tree.sh                    | 2 --
 t/t5562-http-backend-content-length.sh | 2 --
 t/test-lib.sh                          | 4 ++++
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/t/t5000-tar-tree.sh b/t/t5000-tar-tree.sh
index d4730481384..e1fa34bb828 100755
--- a/t/t5000-tar-tree.sh
+++ b/t/t5000-tar-tree.sh
@@ -38,8 +38,6 @@ test_lazy_prereq TAR_NEEDS_PAX_FALLBACK '
 	)
 '
 
-test_lazy_prereq GZIP 'gzip --version'
-
 get_pax_header() {
 	file=$1
 	header=$2=
diff --git a/t/t5562-http-backend-content-length.sh b/t/t5562-http-backend-content-length.sh
index b68ec22d3fd..e83aa336fa8 100755
--- a/t/t5562-http-backend-content-length.sh
+++ b/t/t5562-http-backend-content-length.sh
@@ -3,8 +3,6 @@
 test_description='test git-http-backend respects CONTENT_LENGTH'
 . ./test-lib.sh
 
-test_lazy_prereq GZIP 'gzip --version'
-
 verify_http_result() {
 	# some fatal errors still produce status 200
 	# so check if there is the error message
diff --git a/t/test-lib.sh b/t/test-lib.sh
index 01e88781dd2..33bb9fe991f 100644
--- a/t/test-lib.sh
+++ b/t/test-lib.sh
@@ -1922,6 +1922,10 @@ test_lazy_prereq LONG_IS_64BIT '
 test_lazy_prereq TIME_IS_64BIT 'test-tool date is64bit'
 test_lazy_prereq TIME_T_IS_64BIT 'test-tool date time_t-is64bit'
 
+test_lazy_prereq GZIP '
+	gzip --version
+'
+
 test_lazy_prereq CURL '
 	curl --version
 '
-- 
2.39.1.1392.g63e6d408230

